### PR TITLE
add backfill for setup_interviews permission

### PR DESCRIPTION
**Pull request by richardpattinson**
## Context

Adding the setup_interviews permission to all permissions sets with make_decisions permission. This will enable us to read from it later.

## Link to Trello card

https://trello.com/c/YZFD0yQA/3903-datamigration-to-grant-setupinterview-permissions
## Things to check

- [X] This code does not rely on migrations in the same Pull Request
- [X] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [X] API release notes have been updated if necessary
- [X] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: spec/services/data_migrations/backfill_setup_interviews_permission_spec.rb
Comment (despo) [inline spec/services/data_migrations/backfill_setup_interviews_permission_spec.rb:21]:
```suggestion
  it 'does not update the audit log', with_audited: true do
```
this is required to ensure auditing is enabled

File: app/services/data_migrations/backfill_setup_interviews_permission.rb
Comment (despo) [inline app/services/data_migrations/backfill_setup_interviews_permission.rb:4]: We want to trigger this manually as soon as the flag is enabled 
```suggestion
    MANUAL_RUN = true
```

**Comment by despo** [comment]
Left some minor feedback. The column though should have been `set_up_interviews` rather than `setup_interviews` so we may need to resolve that first.

File: app/services/data_migrations/backfill_setup_interviews_permission.rb
Comment (despo) [inline app/services/data_migrations/backfill_setup_interviews_permission.rb:8]:
```suggestion
        ProviderPermissions.where(make_decisions: true).update_all(set_up_interviews: true)
```

File: spec/services/data_migrations/backfill_setup_interviews_permission_spec.rb
Comment (despo) [inline spec/services/data_migrations/backfill_setup_interviews_permission_spec.rb:4]:
```suggestion
  let!(:provider_permissions) { create(:provider_permissions, make_decisions: true, set_up_interviews: false) }
```

File: spec/services/data_migrations/backfill_setup_interviews_permission_spec.rb
Comment (despo) [inline spec/services/data_migrations/backfill_setup_interviews_permission_spec.rb:8]:
```suggestion
    expect(provider_permissions.reload.set_up_interviews).to be(true)
```

**Comment by despo** [comment]
I've updated the column name, so added some further feedback to reflect that. We could also update the spec descriptions to reflect the new column name, but I'll leave that to you to decide as I don't think it's major given the temporary nature of these files.

File: app/services/data_migrations/backfill_setup_interviews_permission.rb
Comment (richardpattinson) [reply app/services/data_migrations/backfill_setup_interviews_permission.rb]: Cant believe i missed that one =P

**Comment by despo** [comment]
@richardpattinson thanks for all the changes. Let's wait until we deploy the column changes before merging.

**Review by despo — APPROVED** [review]

**Pull request merged**
